### DISCUSSION TITLE
Update Column Definitions

### DISCUSSION THREAD

**Discussion by mg3146** [anvilistas/tabulator#60] [Q&A] [answered]
Hi,

From the Tabulator docs (http://tabulator.info/docs/4.5/columns#replace), there is a way to update column definitions without having to reset them all. 

Is there a way to do the same in Anvil's Tabulator object?  I haven't seen a "updateColumnDefinition" function in the auto completer.

**Comment by lewysigns** [comment, marked as answer]
The auto completer doesn't capture all of the capabilities of the library. I would do the old "try and fail" method to test if the function exist or not. I would also note that IF it is not apart of the auto completer , it generally follows the JavaScript syntax ( meaning camelCase).

I tried said method above and got it work:

```
def update_tabulator(self,**event_args):
   #field='name', changed 'title' attribute to "Updated Title'
   self.tabulator.updateColumnDefinition("name", {'title':"Updated Title"})
```

**Reply by s-cork** [reply]
Any method like that will also work with snake case 😊. 

Autocompletions are just for a limited subset of the available methods (there are quite a lot and each one has to be manually added).

If there are some missing that you use all the time they can easily be added.

**Reply by mg3146** [reply]
Great, thanks!
